### PR TITLE
ruler: Add support for simple relabelling in form of label drop.

### PR DESCRIPTION
**Pull request by bwplotka**
This allows to setup thanos ruler in HA and avoid duplicated alerts.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/rule.go [hunk 1]
```diff
@@ -75,6 +75,9 @@ func registerRule(m map[string]setupFunc, app *kingpin.Application, name string)
 	alertmgrs := cmd.Flag("alertmanagers.url", "Alertmanager URLs to push firing alerts to. The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect Alertmanager IPs through respective DNS lookups. The port defaults to 9093 or the SRV record's value. The URL path is used as a prefix for the regular Alertmanager API path.").
 		Strings()
 
+	alertmgrsExcludeLabels := cmd.Flag("alertmanagers.label-drop", "Labels by name to drop before sending to alertmanager. This allows alert to be deduplicated on replica label (repeated). Similar Prometheus alert relabelling").
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply cmd/thanos/rule.go]: can agree (:

File: pkg/alert/alert.go [hunk 2]
```diff
@@ -179,10 +182,21 @@ func (q *Queue) Push(alerts []*Alert) {
...
+		lb := labels.NewBuilder(labels.Labels{})
+		for _, l := range a.Labels {
+			if q.excludeLabels.Has(l.Name) {
+				continue
+			}
+			lb.Set(l.Name, l.Value)
+		}
+
 		for _, l := range q.labels {
+			if q.excludeLabels.Has(l.Name) {
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply pkg/alert/alert.go]: true, thanks for tip!

**Comment by bwplotka** [comment]
@FUSAKLA PTAL (:

**Comment by FUSAKLA** [comment]
Great :+1:

**Pull request merged**